api/src/routes/settings.test.ts

```diff
@@ -186,6 +186,165 @@ describe('settingRoutes', () => {
       });
     });
 
+    describe('/update-my-username', () => {
+      test('PUT returns an error when the username uses special characters', async () => {
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: 'twaha@'
+        });
+
+        expect(response?.statusCode).toEqual(200);
+
+        expect(response?.body).toEqual({
+          message: 'Username twaha@ contains invalid characters',
+          type: 'info'
+        });
+      });
+
+      test('PUT returns an error when the username is an endpoint', async () => {
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: 'german'
+        });
+
+        expect(response?.statusCode).toEqual(200);
+
+        expect(response?.body).toEqual({
+          message: 'flash.username-taken',
+          type: 'info'
+        });
+      });
+
+      test('PUT returns an error when the username is a bad word', async () => {
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: 'ass'
+        });
+
+        expect(response?.statusCode).toEqual(200);
+
+        expect(response?.body).toEqual({
+          message: 'flash.username-taken',
+          type: 'info'
+        });
+      });
+
+      test('PUT returns an error when the username is a https status code', async () => {
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: '404'
+        });
+
+        expect(response?.statusCode).toEqual(200);
+
+        expect(response?.body).toEqual({
+          message: 'Username 404 is a reserved error code',
+          type: 'info'
+        });
+      });
+
+      test('PUT returns an error when the username is shorter than 3 characters', async () => {
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: 'fo'
+        });
+
+        expect(response?.statusCode).toEqual(400);
+
+        expect(response?.body).toEqual({
+          code: 'FST_ERR_VALIDATION',
+          error: 'Bad Request',
+          message: 'body/username must NOT have fewer than 3 characters',
+          statusCode: 400
+        });
```

Comment:
```suggestion
        expect(response?.body).toEqual({
          message: 'Username at is too short',
          type: 'info'
        });
```
I should have spotted this earlier, but I'm pretty sure this will break the client, due the lack of a `type` property. At the very least, it expects one.

Also, this got me thinking: all the { type: 'info' ...} responses should have code 400, since they're all bad requests.